python/pyarrow/tests/test_table.py

```diff
@@ -735,6 +736,22 @@ def test_table_select_column():
         table.column(4)
 
 
+def test_table_column_with_duplicates():
+    # ARROW-8209
+    table = pa.table([pa.array([1, 2, 3]),
+                      pa.array([4, 5, 6]),
+                      pa.array([7, 8, 9])], names=['a', 'b', 'a'])
+
+    with pytest.raises(KeyError,
+                       match='Field "a" exists 2 times in table schema'):
+        table.column('a')
+
+    assert table.schema.get_field_index('a') == -1
+
+    # Not guaranteed to be sorted
```

Comment:
should we guarantee it in the python API? (eg by just putting a `return sorted(...)` around the return value in `get_all_field_indices`)

Reply:
I may as well sort in the C++ function. Stand by